src/key_extractor.rs

```diff
@@ -1,47 +1,219 @@
+    ///         Err(SimpleKeyExtractionError::new("Extract error"))
+    ///     }
+    ///
+    ///     fn exceed_rate_limit_response(
+    ///             &self,
+    ///             negative: &NotUntil<QuantaInstant>,
+    ///             mut response: HttpResponseBuilder,
+    ///         ) -> HttpResponse {
+    ///             let wait_time = negative
+    ///                 .wait_time_from(DefaultClock::default().now())
+    ///                 .as_secs();
+    ///             response
+    ///                 .content_type(ContentType::plaintext())
+    ///                 .body(format!("Too many requests, retry in {}s", wait_time))
+    ///     }
+    /// }
+    /// ```
+    fn exceed_rate_limit_response(
+        &self,
+        negative: &NotUntil<QuantaInstant>,
+        mut response: HttpResponseBuilder,
+    ) -> HttpResponse {
         let wait_time = negative
             .wait_time_from(DefaultClock::default().now())
             .as_secs();
-        (
-            format!("Too many requests, retry in {}s", wait_time),
-            ContentType::plaintext(),
-        )
+        response
+            .content_type(ContentType::plaintext())
+            .body(format!("Too many requests, retry in {}s", wait_time))
+    }
+
+    #[cfg(feature = "log")]
+    /// Value of the extracted key (only used in logs).
+    fn key_name(&self, _key: &Self::Key) -> Option<String> {
+        None
     }
+}
 
-    /// Error function, will pass [`Self::KeyExtractionError`] to it to return the response error
-    /// when the [`Self::extract`] failed [Read more]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+/// A [KeyExtractor] that allow to do rate limiting for all incoming requests. This is useful if you want to hard-limit the HTTP load your app can handle.
+pub struct GlobalKeyExtractor;
+
+#[derive(Debug)]
+/// A simple struct to create  error, by default the status is  500 server error and content-type is plintext
+pub struct SimpleKeyExtractionError<T: Display + Debug> {
+    pub body: T,
+    pub status_code: StatusCode,
+    pub content_type: ContentType,
+}
+
+impl<T: Display + Debug> SimpleKeyExtractionError<T> {
+    /// Create new instance by body
+    ///
+    /// # Example
+    /// ```rust
+    /// use actix_governor::SimpleKeyExtractionError;
+    /// use actix_http::StatusCode;
+    /// use actix_web::http::header::ContentType;
+    ///
+    /// let my_error = SimpleKeyExtractionError::new("Some error content");
+    ///
+    /// assert_eq!(my_error.body, "Some error content");
+    /// assert_eq!(my_error.content_type, ContentType::plaintext());
+    /// assert_eq!(my_error.status_code, StatusCode::INTERNAL_SERVER_ERROR);
+    /// ```
+    pub fn new(body: T) -> Self {
+        Self {
+            body,
+            status_code: StatusCode::INTERNAL_SERVER_ERROR,
+            content_type: ContentType::plaintext(),
+        }
+    }
+
+    /// Set new status code, the default is [`StatusCode::INTERNAL_SERVER_ERROR`]
```

Comment:
I'd add an "a" to make it sound more like correct English: `/// Set a new status code`
Also for `set_status_code` in line 164